code/modules/mining/ores_coins.dm

```diff
@@ -17,6 +17,27 @@
 			user << "<span class='info'>Not enough fuel to smelt [src].</span>"
 	..()
 
+/obj/item/weapon/ore/Crossed(atom/movable/AM)
+	var/obj/item/weapon/storage/bag/ore/OB
+	var/turf/open/floor/F = get_turf(src)
+	if(loc != F)
+		return ..()
+	if(ishuman(AM))
+		var/mob/living/carbon/human/H = AM
+		for(var/thing in H.get_body_slots())
+			if(istype(thing, /obj/item/weapon/storage/bag/ore))
+				OB = thing
+				break
+	else if(isrobot(AM))
+		var/mob/living/silicon/robot/R = AM
+		for(var/thing in R.module_active)
+			if(istype(thing, /obj/item/weapon/storage/bag/ore))
+				OB = thing
+				break
+	if(OB && istype(F, /turf/open/floor/plating/asteroid/basalt/lava_land_surface))
+		F.attackby(OB, AM)
+	return ..()

```

Comment:
you're calling return ..() twice, try to move things around so you don't use the exact same line twice.